sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -469,14 +508,15 @@ def __init__(self, **kwargs) -> None:
         self._keyvault_client_configs = kwargs.pop("keyvault_client_configs", {})
         self._uses_key_vault = (
             self._keyvault_credential is not None
-            or self._keyvault_client_configs is not None
+            or len(self._keyvault_client_configs) > 0
```

Comment:
I guess it could be, they could set `key vault_client_configs` to None, so the default value isn't used. See the line above. I added back the check, but it still needs to have the len check.